metric_learn/constraints.py

```diff
@@ -79,6 +79,10 @@ def chunks(self, num_chunks=100, chunk_size=2, random_state=None):
     chunks = -np.ones_like(self.known_label_idx, dtype=int)
     uniq, lookup = np.unique(self.known_labels, return_inverse=True)
     all_inds = [set(np.where(lookup == c)[0]) for c in xrange(len(uniq))]
+    max_chunks = np.sum([len(s) // chunk_size for s in all_inds])
+    if max_chunks < num_chunks:
+      raise ValueError(('Not enough examples in each class to form '
+                        '%d chunks of %d examples') % (num_chunks, chunk_size))
```

Comment:
maybe the message can me made more informative for the user by specifying the maximal number of chunks that can actually be constructed